McRequest.cpp

```diff
@@ -0,0 +1,153 @@
+#include "McRequest.h"
+
+#include "Types.h"
+
+#include "MY17_Can_Library.h"
+
+void update_permanent_transmit_request_type(Input_T *input, State_T *state);
+
+void McRequest_update_mcRequest(Input_T *input, State_T *state, Output_T *output) {
+  update_permanent_transmit_request_type(input, state);
+
+  const bool notFinishedSendingRequests = 
+    state->mc_request_state->permanent_transmit_request_type != NO_MOTOR_CONTROLLER_REQUEST;
+  const uint32_t lastRequest_ms = state->mc_request_state->last_permanent_transmit_request_ms;
+  const uint32_t msTicks = input->msTicks;
+  const uint32_t nextPermanentTransmitRequest_ms = lastRequest_ms + msTicks;
+  if ((nextPermanentTransmitRequest_ms < msTicks) && notFinishedSendingRequests) {
+    state->mc_request_state->last_permanent_transmit_request_ms = msTicks;
```

Comment:
in the case where you're referencing a given substructure many times, you can just define a shorthand to access it at the top. something like 

Mc_Request *mc = state->mc_request_state;

and then just reference that. it's basically free from a performance perspective, and saves a lot of typing / useless boilerplate.